kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsTest.java

```diff
@@ -50,6 +55,44 @@ void testRequestTimeoutWaitingForHeaders() {
         httpServer.stop(0);
     }
 
+    static Stream<Arguments> legalEnginePaths() {
+        var uri = URI.create("https://localhost:1234");
+        return Stream.of(
+                Arguments.of("no path", uri, "/v1/transit/"),
+                Arguments.of("slash only", uri.resolve("/"), "/v1/transit/"),
+                Arguments.of("basic", uri.resolve("v1/transit"), "/v1/transit/"),
+                Arguments.of("trailing slash", uri.resolve("v1/transit/"), "/v1/transit/"),
+                Arguments.of("single namespace", uri.resolve("v1/ns1/transit"), "/v1/ns1/transit/"),
+                Arguments.of("many namespaces", uri.resolve("v1/ns1/ns2/transit"), "/v1/ns1/ns2/transit/"),
+                Arguments.of("non standard engine name", uri.resolve("v1/mytransit"), "/v1/mytransit/"));
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource
+    void legalEnginePaths(String name, URI uri, String expected) {
```

Comment:
If I was to see a test failure for `legalEnginePaths` it wouldn't be clear what was going wrong. Something like `shouldAcceptUrlPath` makes it clear that we expect the arguments to be accepted by our code.